src/utils/didWebGenerator.ts

```diff
@@ -0,0 +1,229 @@
+import { Agent, KeyType, TypedArrayEncoder } from '@credo-ts/core'
+import fs from 'fs/promises'
+import { JWK } from 'ts-jose'
+import { Env } from '../env.js'
+import PinoLogger from './logger.js'
+
+// Following the format we used for testing in the past
+export interface DidWebDocument {
+  '@context': string[]
+  id: string
+  verificationMethod: Array<{
+    id: string
+    type: string
+    controller: string
+    publicKeyJwk: {
+      kty: string
+      crv: KeyType.Ed25519
+      x: string
+    }
+  }>
+  authentication: string[]
+  assertionMethod: string[]
+  service: Array<{
+    id: string
+    type: string
+    priority: number
+    recipientKeys: string[]
+    routingKeys: string[]
+    serviceEndpoint: string
+  }>
+}
+
+export interface DidWebGenerationResult {
+  did: string
+  didDocument: DidWebDocument
+  privateKey: string
+}
+
+export class DidWebDocGenerator {
+  private agent: Agent
+  private env: Env
+  private logger: PinoLogger
+
+  constructor(agent: Agent, env: Env, logger: PinoLogger) {
+    this.agent = agent
+    this.env = env
+    this.logger = logger
+  }
+
+  /**
+   * Validates if a DID ID follows the did:web specification
+   */
+  private validateDidWebId(didId: string): boolean {
+    // did:web format: did:web:domain
+    const didWebRegex = /^did:web:[a-zA-Z0-9.-]+(?::[a-zA-Z0-9._~:/?#[\]@!$&'()*+,;=%]+)?$/
+    return didWebRegex.test(didId)
+  }
+  /**
+   * Validates if a service endpoint is a valid localhost URL with optional percent-encoded port
+   */
+  private validateServiceEndpoint(endpoint: string): boolean {
+    // Allow http(s)://localhost or http(s)://localhost%3Aport
+    const regex = /^https?:\/\/localhost(%3A\d+)?(\/.*)?$/
+    // Disallow raw colon in localhost:port
+    if (/^https?:\/\/localhost:\d+/.test(endpoint)) return false
+    return regex.test(endpoint)
+  }
+
+  /**
+   * Generates a DID:web document with cryptographic material
+   */
+  async generateDidWebDocument(): Promise<DidWebGenerationResult> {
+    const didId = this.env.get('DID_WEB_ID')
+    const serviceEndpoint = this.env.get('DID_WEB_SERVICE_ENDPOINT')
+
+    // Validate inputs
+    if (!didId) {
+      throw new Error('DID_WEB_ID environment variable is required')
+    }
+    if (!serviceEndpoint) {
+      throw new Error('DID_WEB_SERVICE_ENDPOINT environment variable is required')
+    }
+
+    if (!this.validateDidWebId(didId)) {
+      throw new Error(`Invalid DID:web ID format: ${didId}`)
+    }
+    if (!this.validateServiceEndpoint(serviceEndpoint)) {
+      throw new Error(`Invalid service endpoint format: ${serviceEndpoint}`)
+    }
```

Comment:
the envs could be changed to fn arguments for `generateDidWebDocument()` (and `generateAndRegisterIfNeeded()`), then you don't need to fiddle with envs in the tests, can directly pass the arguments